### PR TITLE
Add pkcs12 license and update retry logic

### PR DESCRIPTION
**Pull request by ChaitanyaKulkarni28**
Add pkcs12 license and update retry logic to not retry on non-retriable errors.

### CODE REVIEW

File: metadata/metadata.go [hunk 1]
```diff
@@ -273,12 +273,26 @@ func (c *Client) updateEtag(resp *http.Response) bool {
...
+}
+
 func (c *Client) retry(ctx context.Context, cfg requestConfig) (string, error) {
 	for i := 1; i <= backoffAttempts; i++ {
 		resp, err := c.do(ctx, cfg)
 
-		// If the context was canceled just return the error and don't retry.
-		if err != nil && errors.Is(err, context.Canceled) {
+		// Check if error is retriable, if not just return the error and don't retry.
+		if err != nil && !shouldRetry(resp, err) {
```
Comment (dorileo) [inline metadata/metadata.go:295]: It should fix one of the issues I noticed, but additionally we want ScheduleJob() to not block the caller, it all should run in its own go routine - with this new non retry'able logic we'll see the symptom go away but we may stumble on it again later when the ShouldEnable() depends on a long lasting operation. We can address it in a follow up PR though.

File: metadata/metadata.go [hunk 2]
```diff
@@ -289,7 +303,15 @@ func (c *Client) retry(ctx context.Context, cfg requestConfig) (string, error) {
 			continue
 		}
 
-		return resp, nil
+		defer resp.Body.Close()
+		md, err := io.ReadAll(resp.Body)
+		if err != nil {
+			logger.Errorf("Failed to read metadata server response bytes: %+v", err)
```
Comment (dorileo) [inline metadata/metadata.go:309]: Since we are here, can you turn this Errorf into a Debugf? And compute a summary to display on the last line?

Comment (ChaitanyaKulkarni28) [reply metadata/metadata.go]: done

File: metadata/metadata.go (hunk 1 above)
Comment (ChaitanyaKulkarni28) [reply metadata/metadata.go:296]: Considering the purpose of `ShouldEnable()` it should not be a long running operation to have a noticeable block. Updated to run as separate go routine and not block agent main.

**Pull request merged**